### PR TITLE
Add send_alert method to ssl lib

### PR DESCRIPTION
**Pull request by tshmul**
Add send_alert method to ssl lib

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by richsalz** [comment]
needs documentation.

**Comment by tshmul** [comment]
Added documentation. let me know if something else is missing

**Comment by mattcaswell** [comment]
Hmm. I'm not sure this is a good idea. Why do you need this?

**Comment by tshmul** [comment]
I'm implementing credssp RDP server, and as stated in https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/29b260de-f102-49ab-96d0-2172f7d3c816, it requires tls alerts to alert login failures cases.

**Comment by mattcaswell** [comment]
> I'm implementing credssp RDP server, and as stated in https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/29b260de-f102-49ab-96d0-2172f7d3c816, it requires tls alerts to alert login failures cases.

Yeah....ok...seems like a good reason to want it.

**Comment by mattcaswell** [comment]
You've labelled this as CLA trivial. Unfortunately I think this goes beyond our definition of such things. You will need to submit a CLA for this:

https://www.openssl.org/policies/cla.html

File: doc/man3/SSL_send_alert.pod [hunk 1]
```diff
@@ -0,0 +1,46 @@
+=pod
+
+=head1 NAME
+
+SSL_send_alert - send ssl alert to ssl connection
```
Comment (mattcaswell) [inline doc/man3/SSL_send_alert.pod:5]: send an SSL/TLS alert to the peer

File: doc/man3/SSL_send_alert.pod [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ int SSL_send_alert(SSL * ssl, int level, int desc);
```
Comment (mattcaswell) [inline doc/man3/SSL_send_alert.pod:11]: Nit: no space after *

File: doc/man3/SSL_send_alert.pod [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+=head1 DESCRIPTION
+
+SSL_send_alert() sends alert to the ssl connection, and terminates session in case level is B<SSL3_AL_FATAL>.
```
Comment (mattcaswell) [inline doc/man3/SSL_send_alert.pod:15]: Please describe the valid values for `level` and `desc`.

File: doc/man3/SSL_send_alert.pod [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+=head1 RETURN VALUES
+
+The following return values can occur:
+
+=over 4
+
+=item E<gt> 0
+
+The send alert operation was succeeded, the return value is the number of bytes actually written to the TLS/SSL connection. (see L<SSL_write(3)>)
```
Comment (mattcaswell) [inline doc/man3/SSL_send_alert.pod:25]: s/succeeded/successful/

Comment (mattcaswell) [reply doc/man3/SSL_send_alert.pod:25]: It does not seem very useful to know the number of bytes written. Perhaps we should just return 1 on success.

File: doc/man3/SSL_send_alert.pod [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+=item Z<><= 0
+
+The send alert operation was not successful, because either the connection was closed, an error occurred or desc is unknown.
```
Comment (mattcaswell) [inline doc/man3/SSL_send_alert.pod:29]: When referring to variable names please enclose in an `I` tag, e.g. `I<desc>`

File: ssl/ssl_lib.c [hunk 6]
```diff
@@ -5867,6 +5867,11 @@ void SSL_set_allow_early_data_cb(SSL *s,
     s->allow_early_data_cb_data = arg;
 }
 
+int SSL_send_alert(SSL * s, int level, int desc)
```
Comment (mattcaswell) [inline ssl/ssl_lib.c:5870]: Please provide a test for this.

**Pull request closed without merging**